src/server/editorServices.ts

```diff
@@ -870,6 +872,8 @@ namespace ts.server {
                     watchDirectory: returnNoopFileWatcher,
                 } :
                 getWatchFactory(this.host, watchLogLevel, log, getDetailWatchInfo);
+
+            this.pnpWatcher = this.watchPnpFile();

```

Comment:
It actually seems strange to code something so specific to pnp generically into this location.